src/meta/meta_ckpt.c

```diff
@@ -616,59 +616,48 @@ __ckpt_load(WT_SESSION_IMPL *session, WT_CONFIG_ITEM *k, WT_CONFIG_ITEM *v, WT_C
 }
 
 /*
- * __wt_metadata_set_base_write_gen --
- *     Set the connection's base write generation.
+ * __wt_metadata_update_base_write_gen --
+ *     Update the connection's base write generation.
  */
 int
-__wt_metadata_set_base_write_gen(WT_SESSION_IMPL *session, uint64_t max_write_gen)
+__wt_metadata_update_base_write_gen(WT_SESSION_IMPL *session, const char *config)
 {
     WT_CKPT ckpt;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
 
-    WT_RET(__wt_meta_checkpoint(session, WT_METAFILE_URI, NULL, &ckpt));
-
-    /*
-     * We track the maximum page generation we've ever seen, and I'm not interested in debugging
-     * off-by-ones.
-     */
-    S2C(session)->base_write_gen = WT_MAX(ckpt.write_gen, max_write_gen) + 1;
+    conn = S2C(session);
+    memset(&ckpt, 0, sizeof(ckpt));
 
-    __wt_meta_checkpoint_free(session, &ckpt);
+    if ((ret = __wt_ckpt_last(session, config, &ckpt)) == 0) {
```

Comment:
`__wt_ckpt_last` is no longer called from outside this file, I'd suggest renaming to `__ckpt_last`.

Reply:
Fixed